From d1443469e5fdf08a15a3cd99033464621a5aa0ee Mon Sep 17 00:00:00 2001
From: Akash <akashniki@gmail.com>
Date: Sun, 17 May 2020 10:06:20 +0200
Subject: [PATCH 1/2] soong: Disable sandbox

AOSP compiler seems to ignore it anyhow. Disabling it silences the following warning:

**Build sandboxing disabled due to nsjail error.**

Signed-off-by: Albert I <kras@raphielgang.org>
---
 ui/build/sandbox_linux.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ui/build/sandbox_linux.go b/ui/build/sandbox_linux.go
index 2de772b0..4874c157 100644
--- a/ui/build/sandbox_linux.go
+++ b/ui/build/sandbox_linux.go
@@ -33,7 +33,7 @@ type Sandbox struct {
 var (
 	noSandbox    = Sandbox{}
 	basicSandbox = Sandbox{
-		Enabled: true,
+		Enabled: false,
 	}
 
 	dumpvarsSandbox = basicSandbox
-- 
2.30.1


From eec0e1b877296c9ac32ae01e7fb53577e7f90c58 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Thu, 23 Apr 2020 16:34:04 -0400
Subject: [PATCH 2/2] use SCHED_BATCH for the build

Signed-off-by: Albert I <kras@raphielgang.org>
---
 soong_ui.bash | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/soong_ui.bash b/soong_ui.bash
index c1c236b9..c2b21aaa 100755
--- a/soong_ui.bash
+++ b/soong_ui.bash
@@ -14,6 +14,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+chrt -bp 0 $$
+
 # To track how long we took to startup. %N isn't supported on Darwin, but
 # that's detected in the Go code, which skips calculating the startup time.
 export TRACE_BEGIN_SOONG=$(date +%s%N)
-- 
2.30.1

